### ISSUE TITLE
TypeError: unsupported operand type(s) for |: 'sip.wrappertype' and 'type'

### ISSUE THREAD

**Issue by geraldo** [opengisch/qpip#27]
On QGIS 3.28.6 with Python 3.9.5 on Windows 10 the following error happens on installing the plugin:

> TypeError: unsupported operand type(s) for |: 'sip.wrappertype' and 'type' 
> Traceback (most recent call last):
>   File "C:\OSGeo4W/apps/qgis-ltr/./python\qgis\utils.py", line 423, in _startPlugin
>     plugins[packageName] = package.classFactory(iface)
>   File "C:\Users/havargas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\a00_qpip\__init__.py", line 2, in classFactory
>     from .plugin import Plugin
>   File "C:\OSGeo4W/apps/qgis-ltr/./python\qgis\utils.py", line 888, in _import
>     mod = _builtin_import(name, globals, locals, fromlist, level)
>   File "C:\Users/havargas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\a00_qpip\plugin.py", line 20, in 
>     class Plugin:
>   File "C:\Users/havargas/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\a00_qpip\plugin.py", line 134, in Plugin
>     def check_deps(self, additional_plugins=[]) -> MainDialog | bool:
> TypeError: unsupported operand type(s) for |: 'sip.wrappertype' and 'type'

**Comment by m-kuhn** [comment]
Are you able to validate if `Union` works for you?

```
from typing import Union

...

def check_deps(self, additional_plugins=[]) -> Union[MainDialog, bool]:
```

**Comment by geraldo** [comment]
I´m not familiar with `Union`, but it seems to work on that machine. At least the following code runs without error and prints out `x`:

```
from typing import Union

def f(x: str) -> Union[str, None]:
    if x:
        return "x"

print(f("ok"))
```

**Comment by jmkerloch** [comment]
I confirm this is fixing the issue on 3.28.1 with python 3.9.5 on Windows

**Comment by jmkerloch** [comment]
I created a PR for the fix #28

**Comment by Danaozhong** [comment]
I got the same problem with the latest release and QGIS 3.34, any chance you could do a new release soon?
